nmtwizard/preprocess/sampler.py

```diff
@@ -373,4 +384,4 @@ def _select_lines(f):
 
         _select_lines(f)
 
-    return all_files.values(), summary
+    return all_files.values(), summary, add_example_weights
```

Comment:
I think we can avoid this additional return value. Before the return you could do:

```python
if add_example_weights:
    for f in all_files.values():
        if not f.oversample_as_weights:
            f.oversample_as_weights = True
            f.oversample = 1
```

What do you think?

Reply:
Last question: maybe I missed it but where do we ensure that the weight is 1 for files using `*Ns`?

Reply:
1 is the default weigh when initializing a SamplerFile : https://github.com/OpenNMT/nmt-wizard-docker/blob/54e6e6bb3b307683b02f563c5a2e56eb3a468c49/nmtwizard/preprocess/sampler.py#L34

Does this answer your question ?

Reply:
But for files that should be oversampled (syntax `*Ns`) the attribute is changed here:

https://github.com/OpenNMT/nmt-wizard-docker/blob/54e6e6bb3b307683b02f563c5a2e56eb3a468c49/nmtwizard/preprocess/sampler.py#L327

So the question is when does it go back to 1 when weights should be saved, but the file used oversampling instead of overweighting?

Reply:
Yes, you are right again, I fixed it.
Maybe I should add a test case for this case (tomorrow :)) ?

Reply:
Yes a test case would be good to avoid surprises. Thanks!